### ISSUE TITLE
Initializing Mojarra with "version.string" and JSF1029 problem

### ISSUE THREAD

**Issue by JCSCatena** [joinfaces/joinfaces#814]
Hi,

We have an application developed with:
- JoinFaces 4.3.0 
- Spring Boot 2.3.0 
- Tomcat Embebed 9.0.35
and when we start the application I notice that the following problem appears in the log:

```12:17:38.041  INFO 29610 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Fetching config from server at : http://xxxxxx.xx.xx:9081
12:17:38.590  INFO 29610 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Located environment: name=APP_frontend, profiles=[v01], label=master, version=xxxxxxxxxxxxxx, state=null
12:17:38.591  INFO 29610 --- [           main] b.c.PropertySourceBootstrapConfiguration : Located property source: [BootstrapPropertySource {name='bootstrapProperties-configClient'}, BootstrapPropertySource {name='bootstrapProperties-http://www.xxx.xxxxxxx.xx/gitlab/configuracion/pre.git/xxx_frontend/v01/application.properties'}]
12:17:38.668  INFO SIN_USUARIO [main] --- e.j.organization.app.AppFrontendApplication    : The following profiles are active: default
12:17:41.129  INFO SIN_USUARIO [main] --- o.s.cloud.context.scope.GenericScope     : BeanFactory id=6b28fcd7-b385-30b6-8506-be3e5d723520
12:17:41.441  INFO SIN_USUARIO [main] --- trationDelegate$BeanPostProcessorChecker : Bean 'org.joinfaces.autoconfigure.javaxfaces.JsfBeansAutoConfiguration$Jsf2_3AutoConfiguration' of type [org.joinfaces.autoconfigure.javaxfaces.JsfBeansAutoConfiguration$Jsf2_3AutoConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
12:17:41.444  INFO SIN_USUARIO [main] --- trationDelegate$BeanPostProcessorChecker : Bean 'org.joinfaces.autoconfigure.javaxfaces.JsfBeansAutoConfiguration' of type [org.joinfaces.autoconfigure.javaxfaces.JsfBeansAutoConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
12:17:41.987  INFO SIN_USUARIO [main] --- o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 9456 (http)
12:17:42.006  INFO SIN_USUARIO [main] --- o.apache.catalina.core.StandardService   : Starting service [Tomcat]
12:17:42.007  INFO SIN_USUARIO [main] --- org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.35]
12:17:48.976  INFO SIN_USUARIO [main] --- org.apache.jasper.servlet.TldScanner     : At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
12:17:49.481  INFO SIN_USUARIO [main] --- o.a.c.c.C.[.[localhost].[/app-web]       : Initializing Spring embedded WebApplicationContext
12:17:49.482  INFO SIN_USUARIO [main] --- o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 10767 ms
12:17:49.806 DEBUG SIN_USUARIO [main] --- e.j.organization.app.AppFrontendApplication    : Tema seteado a none
12:17:51.460  INFO SIN_USUARIO [main] --- o.j.autoconfigure.ClasspathScanUtil      : Loading prepared scan result took 1650.152238ms
12:17:51.560  INFO SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : Initializing Mojarra |version.string| for context '/app-web'
12:17:52.261  INFO SIN_USUARIO [main] --- j.e.r.webcontainer.jsf.application       : JSF1048: PostConstruct/PreDestroy annotations present.  ManagedBeans methods marked with these annotations will have said annotations processed.
12:17:53.737  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.aliasbean.AliasBeanTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.739  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.aliasbean.AliasBeansScopeTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.741  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.calendar.HtmlInputCalendarTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.743  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.date.HtmlInputDateTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.745  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.schedule.ScheduleTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.747  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.tabbedpane.HtmlPanelTabbedPaneTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.750  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.custom.tree.taglib.TreeTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.752  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.validator.ValidatorBaseTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.754  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.validator.ValidatorBaseTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.756  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.validator.ValidatorBaseTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.758  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class 'org.apache.myfaces.validator.ValidatorBaseTagHandler' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.760  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class '[handler-class: null]' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.762  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class '[handler-class: null]' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.764  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class '[handler-class: null]' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.767  WARN SIN_USUARIO [main] --- j.e.resource.webcontainer.jsf.config     : JSF1029: Application is versioned at 2.0 (either explicitly by the version of /WEB-INF/faces-config.xml or the lack of a /WEB-INF/faces-confg.xml), however class '[handler-class: null]' depends on legacy  facelet class. The facelet artifact represented by this class will not be registered.
12:17:53.947  INFO SIN_USUARIO [main] --- .w.PostConstructApplicationEventListener : Running on PrimeFaces 8.0
12:17:53.954  INFO SIN_USUARIO [main] --- .a.PostConstructApplicationEventListener : Running on PrimeFaces Extensions 8.0.2
```
On line 15 the following message appears:

> Initializing Mojarra **|version.string|** for context '/app-web'

It can be seen that the JSF version is not shown and I don't know if it could be due to a joinfaces BUG or that we have forgotten some configuration.

On the other hand, several messages with the code JSF1029 appear in the log. I have the same problem as the previous one, and it is that I don't know if we have correctly configured the project.

The application seems to work correctly, but I do not like these two details and I prefer that if it is a BUG, ​​joinfaces can be resolved.

Right now we are with the versions that I have indicated previously, but previously we had also tested with Spring Boot (2.1.9, 2.2.1, 2.2.5 and currently with 2.3.0). In the case of Joinfaces, we had tested it with versions 4.1.1, 4.1.5 and currently with version 4.3.0) and in all these combinations, we always ran into the same problem.

Our project uses the faces-config.xml file ( in src\main\resources\META-INF) and I paste part of it in this incident, in case this configuration could be the problem.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE xml>
<faces-config
    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-facesconfig_2_2.xsd"
    version="2.2">
	<navigation-rule>
		<from-view-id>*</from-view-id>

......
......
......


	<application>
		<el-resolver>org.springframework.web.jsf.el.SpringBeanFacesELResolver
		</el-resolver>
		<locale-config>
			<default-locale>es</default-locale>
		</locale-config>
		<resource-bundle>
			<base-name>textos</base-name>
			<var>textos</var>
	   	</resource-bundle>
	</application>
</faces-config>
```

We have also tried to put version 2.3 in the faces-config making the following change:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE xml>
<faces-config
    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-facesconfig_2_3.xsd"
    version="2.3">
```

But we get the same result.

Can this be a bug?

Regards, and thank you very much.

**Comment by larsgrefer** [comment]
Can you provide an example project, which reproduces the issue?

**Comment by JCSCatena** [comment]
Ok, I prepare a frontend with the minimum so you can test it.

We have uploaded to Joinfaces version 4.3.3 and the problem continues to occur.

Regards, and thank you very much.

**Comment by JCSCatena** [comment]
Hi,

I have already created the example frontend and one of the possible errors that I mentioned (the appearance of the message: **JSF1029**) was due to a library of myfaces version JSF1.2 that I needed to incorporate into the project, and since the project actually uses JSF 2.3 , towards that when including this library this message will be displayed. I have removed the myfaces library from the project and the message has disappeared.

Now there is only the other possible BUG, ​​and that is that it still does not show me the version of Mojara, the following message appears: "**Initializing Mojarra | version.string | for context** '/ app-web'".

In order to launch the Frontend project, the following three steps are required, which are indicated in the README.txt file:

STEP 1: Install lib in your local repository (need one library)
include_library.bat

STEP 2: Compile de Frontend Proyect
mvn clean package -DskipTests

STEP 3: Run de Frontend (You can see the message of the mojarra version)
start_frontend.bat

Example proyect
[Frontend_example.zip](https://github.com/joinfaces/joinfaces/files/5338713/Frontend_example.zip)

Regards, and thank you very much.

**Comment by marcelocaser** [comment]
Yes.. i have the same problem in my project!!

**Comment by larsgrefer** [comment]
The version number not beeing printed correctly is a bug in mojarra itself: https://github.com/eclipse-ee4j/mojarra/issues/4638

This seems to be fixed in Mojarra 2.3.15

**Comment by JCSCatena** [comment]
version 2.3.15 was released a while ago.  Does anyone know how I can incorporate it into my application?  

Greetings and thank you
